Modules/Devices/src/Devices.Infrastructure/PushNotifications/DirectPush/ApplePushNotificationService/ApplePushNotificationServiceConnector.cs

```diff
@@ -60,20 +64,23 @@ public void ValidateRegistration(PnsRegistration registration)
             throw new InfrastructureException(InfrastructureErrors.InvalidPushNotificationConfiguration(_options.GetSupportedBundleIds()));
     }
 
-    private async Task HandleResponse(HttpResponseMessage response, string handle)
+    private async Task HandleResponse(HttpResponseMessage response, PnsRegistration registration)
     {
         if (response is { IsSuccessStatusCode: false })
         {
-            var responseContent = await response.Content.ReadAsStringAsync();
-            if (!responseContent.IsNullOrEmpty())
+            var responseContent = JsonConvert.DeserializeObject<ErrorResponse>(await response.Content.ReadAsStringAsync());
+            if (!responseContent.Reason.IsNullOrEmpty())
             {
                 _logger.LogError(
-                    "The following error occurred while trying to send the notification for handle '{handle}': '{responseContent}'",
-                    handle, responseContent);
+                    "The following error occurred while trying to send the notification for deviceId '{deviceId}': '{error}'",
+                    registration.DeviceId, responseContent.Reason);
```

Comment:
In case of the "Unregistered" response you should not log an error. This only causes confusion when looking into the logs. You should only log with level "information". But in all other cases, you can log an error.